src/main.cpp

```diff
@@ -3193,8 +3202,14 @@ bool ReceivedBlockTransactions(const CBlock &block, CValidationState& state, CBl
                 } else {
                     pindex->nChainSproutValue = boost::none;
                 }
+                if (pindex->pprev->nChainSaplingValue) {
+                    pindex->nChainSaplingValue = *pindex->pprev->nChainSaplingValue + pindex->nSaplingValue;
+                } else {
+                    pindex->nChainSaplingValue = boost::none;
```

Comment:
In the 'Sprout' block above we are also checking ```&& pindex->nSproutValue```. If this is false we are setting ```pindex->nChainSproutValue = boost::none;```. I see that the nSalplingValue is not optional, but why do we not have an equivalent branch here? Does the nChainSaplingValue ever get set to none?

Reply:
For Sprout, we added value pool tracking after launch, so most nodes did not have recorded values for old indices. Thus we had to make `nSproutValue` optional, and handle its potential absence. For this particular branch, that means we only set `nChainSaplingValue` for a block if we have both `nChainSaplingValue` for its parent, *and* `nSaplingValue` for the block (this is the only case out of four possible that is valid).

For Sapling, we know that all previous blocks by definition have `nSaplingValue = 0`, and thus as long as we merge this before Sapling activates, everyone's nodes will have a correct `nSaplingValue` for every block. So we only need to check that we have the parent's `nChainSproutValue` (which might still not be the case, if this block is an orphan).